s3/s3_operations.cpp

```diff
@@ -78,6 +78,7 @@ namespace irods_s3 {
         std::ios_base::openmode open_mode;
         std::shared_ptr<dstream> dstream_ptr;
         std::shared_ptr<s3_transport> s3_transport_ptr;
+        bool checked_for_checksum_read_after_put;
```

Comment:
Consider initializing the boolean.

Reply:
Are you asking for me to add a constructor / initialization list for this struct?

Reply:
No, you don't have to add a constructor. You can initialize the bool directly.

You may want to initialize the open_mode too.

Keep in mind I don't how the struct is handled in the plugin. If you can guarantee that the struct is never in an invalid state, then maybe none of this is necessary.